quinn-proto/src/connection/mod.rs

```diff
@@ -2565,15 +2559,17 @@ where
     }
 
     /// Issue an initial set of connection IDs to the peer
-    fn issue_cids(&mut self) {
-        if self.local_cid_len == 0 {
+    fn issue_cids(&mut self, now: Instant) {
+        if self.local_cid_state.cid_len() == 0 {
             return;
         }
 
         // Subtract 1 to account for the CID we supplied while handshaking
         let n = self.peer_params.issue_cids_limit() - 1;
         self.endpoint_events
-            .push_back(EndpointEventInner::NeedIdentifiers(n));
+            .push_back(EndpointEventInner::NeedIdentifiers(now, n));
+        // Track lifetime of cid used in handshaking
+        self.local_cid_state.track_lifetime(0, now);
```

Comment:
I don't think the difference between the time the `Connection` is created and the time when CIDs are issued is significant enough to justify doing this here. Instead, let's set up the state for the handshake CID in `CidState::new`, and thereby encapsulate `CidState` a bit more tightly.

Reply:
Agree.

Reply:
I recall why i put it here in the first place. It was because UT uses artificial time and thus we have to pass `now` to this function to track the timeline. Moving this to `new` of `CidState` will make `CidState` also dependent on `now`.

Reply:
Is that a problem? `now` is already available in `Connection::new`, where `CidState` would be constructed.

Reply:
Right. I didn't finish the response. No issue to move forward :-)